File: algorithm-join-point-2/src/main/java/ru/vood/joinpoint2/infrastructure/bean/WorkerBean2.java
package ru.vood.joinpoint2.infrastructure.bean;

import org.springframework.stereotype.Component;
import ru.vood.joinpoint2.infrastructure.flow.context.Bean2ReturnContext;
import ru.vood.joinpoint2.infrastructure.flow.context.Bean2RunContext;

@Component("Bean2")
public class WorkerBean2 extends AbstractWorkerBean<Bean2RunContext, Bean2ReturnContext> {


    public WorkerBean2() {
        super(Bean2RunContext.class, Bean2ReturnContext.class);
    }


    @Override
    public Bean2ReturnContext doIt(Bean2RunContext bean2RunContext) {
        return new Bean2ReturnContext(Long.valueOf(bean2RunContext.getCnt()));
    }
}

File: algorithm-join-point-2/src/main/java/ru/vood/joinpoint2/infrastructure/bean/WorkerBean2.kt
package ru.vood.joinpoint2.infrastructure.bean

import org.springframework.stereotype.Component
import ru.vood.joinpoint2.infrastructure.flow.context.Bean2ReturnContext
import ru.vood.joinpoint2.infrastructure.flow.context.Bean2RunContext


@Component("Bean2")
class WorkerBean2 : AbstractWorkerBean<Bean2RunContext, Bean2ReturnContext>(Bean2RunContext::class.java, Bean2ReturnContext::class.java) {

    override fun doIt(bean2RunContext: Bean2RunContext): Bean2ReturnContext {
        return Bean2ReturnContext(java.lang.Long.valueOf(bean2RunContext.cnt))
    }
}
